commandapi-core/src/main/java/dev/jorel/commandapi/StringTooltip.java

```diff
@@ -28,8 +35,58 @@
 public class StringTooltip implements IStringTooltip {
 
 	private final String suggestion;
-	private final String tooltip;
-	
+	private final Message tooltip;
+
+
+
+	/**
+	 * Constructs a StringTooltip with a suggestion and a string tooltip
+	 *
+	 * @param suggestion the suggestion to provide to the user
+	 * @param tooltip    the string tooltip to show to the user when they hover over the
+	 *                   suggestion
+	 * @return a StringTooltip representing this suggestion and tooltip
+	 */
+	public static StringTooltip ofString(String suggestion, String tooltip) {
+		return ofMessage(suggestion, Tooltip.toMessage(tooltip));
+	}
+
+	/**
+	 * Constructs a StringTooltip with a suggestion and a formatted tooltip
+	 *
+	 * @param suggestion the suggestion to provide to the user
+	 * @param tooltip    the formatted tooltip to show to the user when they hover over the
+	 *                   suggestion
+	 * @return a StringTooltip representing this suggestion and tooltip
+	 */
+	public static StringTooltip ofMessage(String suggestion, Message tooltip) {
+		return tooltip == null ? none(suggestion) : new StringTooltip(suggestion, tooltip);
+	}
+
+	/**
+	 * Constructs a StringTooltip with a suggestion and a formatted bungee text component tooltip
+	 *
+	 * @param suggestion the suggestion to provide to the user
+	 * @param tooltip    the formatted tooltip to show to the user when they hover over the
+	 *                   suggestion
+	 * @return a StringTooltip representing this suggestion and tooltip
+	 */
+	public static StringTooltip ofBungeeComponents(String suggestion, BaseComponent... tooltip) {
```

Comment:
I'd rather the method name be called `ofBaseComponents` rather than `ofBungeeComponents` - a name reflecting the class name is definitely much more preferable. The CommandAPI doesn't support Bungee and the naming is misleading and possibly also confusing for Spigot developers that haven't used Bungee.

This also applies to `Tooltip.java` and the `ofBungeeComponents` and `generateBungeeComponents` methods

Reply:
d4bc110